pkg/ride/tree_evaluator.go

```diff
@@ -384,6 +388,36 @@ func (e *treeEvaluator) walk(node Node) (rideType, error) {
 		for i, arg := range args {
 			e.s.cs[len(e.s.cs)-1][i] = arg
 		}
+
+		if n.Name == "callDApp" {
+			addr := args[0].value
+			funcName := args[1].value
+			recipient, err := extractRecipient(addr)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to get recipient from addr, user function '%s", id)
+			}
+
+			args := make([]rideType, len(n.Arguments))
+			for i, arg := range n.Arguments {
+				a, err := e.walk(arg) // materialize argument
+				if err != nil {
+					return nil, errors.Wrapf(err, "failed to materialize argument %d of system function '%s'", i+1, id)
+				}
+				if isThrow(a) {
+					return a, nil
+				}
+				args[i] = a
+			}
+
+			res, err := invokeFunctionFromDApp(e.env, recipient, funcName, args[2:])
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to get RideResult from invokeFunctionFromDApp of system function '%s'", id)
+			}
+			resultDAppFromDApp = res
```

Comment:
As I told you, check `res` here for an error and add ScriptAcitons to evaluator's actions list in case of no error.